In this notebook, we are going to explore COVID-19 cases in the United States by state.

Import the necessary modules.

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

print("Setup complete.")

Setup complete.


Load the dataset to begin analysis.

In [2]:
us_states_df = pd.read_csv("/kaggle/input/covid19-in-usa/us_states_covid19_daily.csv")

Use the .head() attribute to preview the first few rows of the DataFrame.

In [3]:
us_states_df.head()

date state  positive  probableCases   negative  pending  \
0  20201206    AK   35720.0            NaN  1042056.0      NaN   
1  20201206    AL  269877.0        45962.0  1421126.0      NaN   
2  20201206    AR  170924.0        22753.0  1614979.0      NaN   
3  20201206    AS       0.0            NaN     2140.0      NaN   
4  20201206    AZ  364276.0        12590.0  2018813.0      NaN   

  totalTestResultsSource  totalTestResults  hospitalizedCurrently  \
0        totalTestsViral         1077776.0                  164.0   
1  totalTestsPeopleViral         1645041.0                 1927.0   
2        totalTestsViral         1763150.0                 1076.0   
3        totalTestsViral            2140.0                    NaN   
4  totalTestsPeopleViral         2370499.0                 2977.0   

   hospitalizedCumulative  ...   posNeg  deathIncrease  hospitalizedIncrease  \
0                   799.0  ...  1077776              0                     0   
1                 26331.0  ...  1691003             12                     0   
2                  9401.0  ...  1785903             40                    21   
3                     NaN  ...     2140              0                     0   
4                 28248.0  ...  2383089             25                   242   

                                       hash  commercialScore  \
0  7b1d31e2756687bb9259b29195f1db6cdb321ea6                0   
1  19454ed8fe28fc0a7948fc0771b2f3c846c1c92e                0   
2  25fc83bffff5b32ba1a737be8e087fad9f4fde33                0   
3  8c39eec317586b0c34fc2903e6a3891ecb00469e                0   
4  7cf59da9e4bc31d905e179211313d08879880a85                0   

  negativeRegularScore negativeScore positiveScore score  grade  
0                    0             0             0     0    NaN  
1                    0             0             0     0    NaN  
2                    0             0             0     0    NaN  
3                    0             0             0     0    NaN  
4                    0             0             0     0    NaN  

[5 rows x 55 columns]

We first need inspect the dataset and identify the data types of the variables. Pandas automatically determines variables data types, but sometimes is incorrect. We use the **.info()** method to quickly inspect the statistics for each column, including the data type and number of entries.

In [4]:
us_states_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15633 entries, 0 to 15632
Data columns (total 55 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   date                         15633 non-null  int64  
 1   state                        15633 non-null  object 
 2   positive                     15481 non-null  float64
 3   probableCases                5449 non-null   float64
 4   negative                     15323 non-null  float64
 5   pending                      1684 non-null   float64
 6   totalTestResultsSource       15633 non-null  object 
 7   totalTestResults             15598 non-null  float64
 8   hospitalizedCurrently        12516 non-null  float64
 9   hospitalizedCumulative       9434 non-null   float64
 10  inIcuCurrently               7713 non-null   float64
 11  inIcuCumulative              2700 non-null   float64
 12  onVentilatorCurrently        6211 non-null   float64
 13  onVentilatorCumu

Using this information, we can tell that there are a total of 15,633 entries in the csv file, each involving COVID-19 results. However, many columns in this DataFrame have incorrect data types, and many are missing significant amounts of data. 

We will first correct the data types so that every variable has its correct type.